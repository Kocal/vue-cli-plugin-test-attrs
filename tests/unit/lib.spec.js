const compiler = require('vue-template-compiler')
const fs = require('fs')
const path = require('path')
const template = fs.readFileSync(
  path.join(__dirname, 'resources/template.html'),
  { encoding: 'utf8' }
)

const generateCompilerModule = require('../../lib')

function generateRender(options) {
  const compilerModule = generateCompilerModule(options)

  const { render } = compiler.compile(template, {
    modules: [compilerModule],
  })

  return render
}

describe('Tests', () => {
  it('renders with test attribue when disabled', () => {
    const render = generateRender({
      enabled: false,
      attrs: ['test', 'e2e'],
    })

    expect(render).toEqual(expect.stringContaining(`"data-test":"paragraph"`))
    expect(render).toEqual(expect.stringContaining(`"data-e2e":"div"`))
    expect(render).toMatchSnapshot()
  })

  it('renders without test attribute when enabled', () => {
    const render = generateRender({
      enabled: true,
      attrs: ['test', 'e2e'],
    })

    expect(render).toEqual(
      expect.not.stringContaining(`"data-test":"paragraph"`)
    )
    expect(render).toEqual(expect.not.stringContaining(`"data-e2e":"div"`))
    expect(render).toMatchSnapshot()
  })
})
