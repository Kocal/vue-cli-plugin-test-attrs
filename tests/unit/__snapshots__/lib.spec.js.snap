// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests renders with test attribue when disabled 1`] = `"with(this){return (toggle)?_c('div',[_c('p',{attrs:{\\"data-test\\":\\"paragraph\\"}},[_v(\\"\\\\n    The Paragraph\\\\n  \\")]),_v(\\" \\"),_c('span',{attrs:{\\"data-test\\":\\"span\\"}},[_v(\\"\\\\n    The Span\\\\n  \\")]),_v(\\" \\"),_c('div',{attrs:{\\"data-e2e\\":\\"div\\"}},[_v(\\"\\\\n    The div\\\\n  \\")])]):_e()}"`;

exports[`Tests renders without test attribute when enabled 1`] = `"with(this){return (toggle)?_c('div',[_c('p',[_v(\\"\\\\n    The Paragraph\\\\n  \\")]),_v(\\" \\"),_c('span',[_v(\\"\\\\n    The Span\\\\n  \\")]),_v(\\" \\"),_c('div',[_v(\\"\\\\n    The div\\\\n  \\")])]):_e()}"`;
